### ISSUE TITLE
[PUP] Scanner / Spiritreaver Head - Not Evaluating Resistances

### ISSUE THREAD

**Issue by FariskaEden** [EdenServer/community#1765]
### Checklist

- [X] I have searched for existing issues for issues like this one. The issue has not been posted. (Duplicate reports slow down development.)
- [X] I have provided reproducable steps. (No "as the title says" posts please. Provide reproducable steps even if it seems like a waste of time.)
- [X] I have provided my client version in the details. (type /ver into your game window)

### Details

When either a Scanner or Spiritreaver head are equipped the automaton will still cast spells that are easily resisted by it's target. For example I used the Spiritreaver head versus Suzaku and the automaton would cast Blizzard IV and Fire IV versus it even though it easily resists both those elements. What I believe the automaton does instead is cast its elemental spells from highest skill level to lowest.

https://ffxiclopedia.fandom.com/wiki/Scanner
https://ffxiclopedia.fandom.com/wiki/Spiritreaver_Head

version 30181205_0

**Comment by xizeta** [comment]
After reading through the wiki this is my understanding of the correct behaviour:

* Spiritreaver should avoid highly resisted spells for Elemental nukes
* When scanner is equipped, an Automaton should avoid highly resisted Enfeebles.

**Comment by FariskaEden** [comment]
Tried to do some more testing to double check these points.

- Spiritreaver should avoid highly resisted elements for Elemental nukes.
Correct, currently it will cast elemental nukes that are highly resisted without a Scanner equipped. Tested this versus a lesser colibri and the automaton would cast Aero IV and versus an antlion Stone IV was cast.

- Scanner may cause other automatons to have the same effect.
Correct, after testing this it seems that automaton will avoid casting elemental nukes that are highly resisted even when maneuvers are set up in a way to force the highly resisted spell. I tested this with a Stormwaker head and Spiritweaver head with both having a scanner equipped. For example water elemental nukes were not cast versus crabs, wind elemental nukes were not cast versus colibri and earth elemental nukes were not cast versus antlions. This aspect works correctly.

- When scanner is equipped, an automaton should avoid highly resisted elements for Enfeebles.
I believe this one is correct, however when tested versus lesser colibris silence was cast, crabs poison was cast and antlions slow was cast. What does seem to work correctly with and without a Scanner are immunities because when tested the automaton would not cast poison versus scorpions. This one would probably require retail testing to confirm.

- Scanner also prevents the automaton from casting Silence on non-spellcasting mobs.
I believe this one is also correct, however some properties of it are odd. Currently it won't cast silence on mobs that have no MP (tested on antlions, scorpions and lizards). However, when tested versus crabs and lesser colibri silence was still cast.

**Comment by xizeta** [comment]
Thanks for giving the full details! This kind of thing really helps us to get it right first time.

**Comment by Iced-Latte** [comment]
This is incorrect information on what scanner should be doing in-era and what the heads do now.

Scanner seems to be currently doing its out-of-era function to effect enfeebles but not nukes. In addition to this it makes the BLM head alternate between the 2 highest tier nukes it has regardless of commands. I did extensive testing last night with Pimpin. Regardless of scanner equipped it will cast the 2 highest nukes and simply alternate.

Without Scanner, You can make the BLM head cast stone IV with a stone maneuver, ect. However it will not cast Thunder III with a Thunder maneuver. With a scanner equipped, it should cast Thunder III if a mobs lowest rests is Thunder.

The era-appropriate version of scanner is as follows:
"As of the December 2006 update, the Scanner now works like the description says, and causes the automaton to attempt to cast nukes based on a mob's elemental weakness, if any."
https://ffxiclopedia.fandom.com/wiki/Scanner?oldid=343921

Some of the information given above via Fariska is incorrect because it is out of era.

**Comment by FariskaEden** [comment]
From my understanding we were given the current casting AI for automatons and some of the later buffs to attachments since you can no longer capture the data for past automaton AI and how they behaved with said parts.

**Comment by Iced-Latte** [comment]
Scanner is a confirmed case for in-era function of nuking elemental resistances.

**Comment by Iced-Latte** [comment]
There is an inherent problem with the BLM head if it is using out of era with the "built in scanner". and Scanner not working for elemental nukes.

When you use ice maneuver, it continues the out of era scanner behavior, and only uses the top 2 nukes it has. So we can currently only make it cast StoneIV, Water IV, Aero IV, and Fire IV.

We can not make it cast Blizzard IV or Thunder III. When ice maneuver is used, it acts like a scanner is equipped again and goes back to top 2 nukes (changed depending on merits).

**Comment by FariskaEden** [comment]
The BLM head does not have the built-in scanner, that is one of the reasons why I made this bug report. My second post includes the tests I did and shows that the scanner helps the automaton avoid elemental nukes that would be resisted regardless if the RDM or BLM head is equipped. Could you detail which mobs you did tests versus and when the scanner attachment was used?

**Comment by Iced-Latte** [comment]
The scanner does NOT make the BLM avoid elemental nukes at all. It makes both BLM and RDM just cast the top 2 nukes they have over and over. I tested with Pimpin on tons of mobs, even Puks which are healed by wind. I use a scanner, and it casts Aero IV and heals the puk.

Same with elementals (they don't get healed just resist).

**Comment by Iced-Latte** [comment]
Amended: I have confirmed testing with Fariska! Here are my conclusions:
(Testing done vs T+ elementals)
-  The BLM auto always seems to choose it's 2 highest spells currently.
-  With Scanner equipped, the automaton will avoid elemental/enfeeble resistances

This means scanner is currently using the post-era version. It is avoiding high resistances in both elemental and enfeebling. This is a double edges sword, because it will not attempt to silence an Ice elemental. (Additionally it will not cast aero or ice on an ice elemental). So it always moves "down" it's nuke tier list. 

The list is as follows with merits: BlizzardIV->FireIV->AeroIV->WaterIV->StoneIV->ThunderIII

Because it only chooses it's 2 highest spells, skipping any resist, ThunderIII will never be cast with a scanner. It also cannot seem to be triggered without a scanner.

Thus: For this scanner issue we must determine if we want the OOE "Avoids resist", or the in era "Targets weakness"

Proposition: Since we cannot change the basic nature of the BLM Automaton choosing 2 spells, We should opt for the in-era "Targets weakness". This would allow for 2 things:

1. Thunder III would be cast on mobs weak to thunder.
2. The RDM auto would not avoid casting enfeebles based on elemental resistances.

**Comment by spaceIdiot** [comment]
ran into this bug today facing Seiryu with my team.  BLM puppet cast only stoneIV and WaterIV when nuking the boss when it is supposed to be weak to Ice.  https://ffxiclopedia.fandom.com/wiki/Seiryu

Scanner was equipt

**Comment by FariskaEden** [comment]
Do you have 5/5 into your automaton's skills merit category? It only gets blizzard IV with 5/5 in that.

**Comment by spaceIdiot** [comment]
ah, no, only got 4 points into that right now.  Thank you for the info, will report back once I have that unlocked.

**Comment by KnowOne134** [comment]
going that we are using the OOE version of automatons what would be needed to change for this ticket, lots of back an forth about what is was and what it is
